src/MessageHandler/ActivityPub/Inbox/DeleteHandler.php

```diff
@@ -35,16 +40,21 @@ public function __construct(
 
     public function __invoke(DeleteMessage $message): void
     {
-        try {
-            $actor = $this->activityPubManager->findRemoteActor($message->payload['actor']);
-        } catch (\Exception) {
-            return;
-        }
+        $actor = $this->activityPubManager->findActorOrCreate($message->payload['actor']);
+
+        $id = \is_array($message->payload['object']) ? $message->payload['object']['id'] : $message->payload['object'];
+        $object = $this->apActivityRepository->findByObjectId($id);
 
-        if (\is_array($message->payload['object'])) {
-            $object = $this->apActivityRepository->findByObjectId($message->payload['object']['id']);
-        } else {
-            $object = $this->apActivityRepository->findByObjectId($message->payload['object']);
+        if (!$object) {
+            $user = $this->userRepository->findOneBy(['apProfileId' => $id]);
+            if ($actor instanceof User && $user instanceof User && $actor->apDomain === $user->apDomain) {
```

Comment:
maybe also check on $user not null/empty? Yea those userRepository calls can sometimes return null.

Reply:
OK let's hope so 😂

Reply:
A PHP playground just confirmed it :D